### PR TITLE
Decouple Pinniped from Tanzu Framework

### PR DESCRIPTION
**Pull request by seemiller**
### What this PR does / why we need it

This PR will decouple Pinniped Post Deploy and Tanzu Auth Controller manager components from any dependency on Tanzu Framework.

### Which issue(s) this PR fixes

Fixes #

### Release note

```release-note

```

### CODE REVIEW

**Comment by seemiller** [comment]
@benjaminapetersen We think we're mostly done here, care to take a look? Thanks!

File: pinniped-components/tanzu-auth-controller-manager/README.md [hunk 1]
```diff
@@ -25,7 +25,7 @@ To generate the default pinniped addon secret:
 ```sh
 # The primary use case of this feature is through the top-level Tanzu Framework Makefile, via a command like:
```
Comment (benjaminapetersen) [inline pinniped-components/tanzu-auth-controller-manager/README.md:26]: We can probably change the comment here as well.

File: pinniped-components/tanzu-auth-controller-manager/hack/generate-package-secret.sh [hunk 2]
```diff
@@ -6,7 +6,6 @@
 set -euo pipefail
 
 MY_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
-TF_ROOT="${MY_DIR}/../../.."
```
Comment (benjaminapetersen) [inline pinniped-components/tanzu-auth-controller-manager/hack/generate-package-secret.sh:9]: So there is one more file in this directory, the `run.sh` file.  Several things here:

It still contains a `TF_ROOT` var:
```
TF_ROOT="${MY_DIR}/../../.."
```
It references `<TF_ROOT>/packages/tanzu-auth`.  This is where the CARVEL package for the `tanzu-auth-controller-manager` lives.  

I think we need a conversation about this.

File: pinniped-components/tanzu-auth-controller-manager/hack/generate-package-secret.test.sh [hunk 3]
```diff
@@ -6,7 +6,6 @@
 set -euo pipefail
 
 MY_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
-TF_ROOT="${MY_DIR}/../../.."
```
Comment (benjaminapetersen) [inline pinniped-components/tanzu-auth-controller-manager/hack/generate-package-secret.test.sh:9]: There is also a `generate.sh` sibling that does:

```
# generate RBAC rules
"${TF_ROOT}/hack/tools/bin/controller-gen" paths=./controllers +rbac:roleName=tanzu-auth-controller-manager output:rbac:dir="${TANZU_AUTH_CONFIG_DIR}"
```
So this is still depending on `TF_ROOT`.

Comment (benjaminapetersen) [reply pinniped-components/tanzu-auth-controller-manager/hack/generate-package-secret.test.sh:9]: Pointing out that it is also depositing files into `<TF_ROOT>/packages/tanzu-auth`

File: pinniped-components/tanzu-auth-controller-manager/hack/generate-package-secret.sh (hunk 2 above)
Comment (benjaminapetersen) [reply pinniped-components/tanzu-auth-controller-manager/hack/generate-package-secret.sh:9]: @yharish991 I believe you had stated `<TF_ROOT>/packages` is going to stay put for now, but when Pinniped components move this work will have to be sheduled?

File: pinniped-components/tanzu-auth-controller-manager/hack/generate-package-secret.test.sh (hunk 3 above)
Comment (benjaminapetersen) [reply pinniped-components/tanzu-auth-controller-manager/hack/generate-package-secret.test.sh:9]: Related to above comment https://github.com/vmware-tanzu/tanzu-framework/pull/3253/files#r969852402.

**Review by benjaminapetersen — APPROVED** [review]
Giving the approval given we either:

- follow with a PR to move $TF_ROOT/package/tanzu-auth into $TF_ROOT/pinniped-components
- OR we have a plan for handling this other package in future work.

👍

**Comment by danniel1205** [comment]
Add `do-not-merge/hold` because there might have some minor fixes needed. e.g. `"${TF_ROOT}/hack/tools/bin/controller-gen"`

**Pull request merged**